6_nodejs/week2/app.js

```diff
@@ -0,0 +1,62 @@
+import express from 'express'
+const app = express();
+const port = process.env.PORT || 3000;
+
+// Support parsing JSON requests
+app.use(express.json());
+
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+
+// GET /search
+app.get('/search', (req, res) =>{
+    const query = req.query.q;
+    if(query){ 
+        const filteredData = data.filter(item => { 
+          for (const key in item) { 
+            if (item[key].toString().includes(query)){ 
+              return true
+            }
+          } return false
+        });
+        res.json(filteredData); 
+    } else {
+      res.json(data); 
+    }
+})
+
+// GET /documents/:id
+app.get('/documents/:id', (req, res) => {
+  const id = parseInt(req.params.id);
+  const data = documents.find(item => item.id === id);
+  if (data) {
+    res.json(data);
+  } else {
+    res.status(404).send("Data is not found")
+  }
+});
+
+// POST /search
+app.post('/search', (req, res) => {
+  const query = req.query.q;
+  const fields = req.body.fields;
+
+  if (query && fields) {
+    res.status(400).send("q and fields cannot be provide at the same time.")
+  } else if (query) {
+    const filteredData = data.filter(item =>{
```

Comment:
In this route, I found a few things to improve.
Create a flag variable which by default is equal to false and I would call it something like isFound before filtering data.
While filtering the array in case adding an item matches a search, change it to be true.
After the filtering, you can check if the isFound variable is still equal to false then you can give feedback that will indicate a successful or unsuccessful search. I will provide an example below.